packages/context/src/context.ts

```diff
@@ -295,9 +231,13 @@ export class Context extends EventEmitter {
    * notified for relevant events.
    */
   protected async waitUntilPendingNotificationsDone(timeout?: number) {
-    const count = this.pendingNotifications;
-    if (count === 0) return;
-    await pEvent.multiple(this, 'observersNotified', {count, timeout});
+    if (!this.eventNotifier) return;
+    const errorHandler = () => {
+      // no-op, eventNotifier will report this error as a rejected promise
```

Comment:
This is a bit ugly.

1. Context is listening for ContextEventNotifier `error` event and re-emits it on the context object itself.
2. `pEvent.multiple` installs the error handler on the notifier object, not the context
3. Because there is no error event handler installed on the context, it triggers unhandled error
3. Before that error is detected by Mocha, `waitUntilPendingNotificationsDone` return a promise rejected because of a timeout.

Before my refactoring, `pEvent.multiple` installed the error handler on the context and thus no uncaught error was reported.

If we don't like the solution proposed here, then we can keep the entire implementation of `waitUntilPendingNotificationsDone` in the Context class only, make sure that `observersNotified` are re-emitted on `Context`, and expose `ContextEventNotifier#pendingNotifications` as a public read-only property that can be consumed by `Context` to setup `pEvent.multiple`. Having said that, this alternative approach seems even more complicated than what I have now.

Thoughts?

Reply:
This is a warning sign that composing `ContextEventNotifier` into `Context` as a property might not be the best approach. Have you considered to have `Context` extend from `ContextEventNotifier` or model `ContextEventNotifier` as a mixin?

Reply:
What about simply moving `waitUntilPendingNotificationsDone` into `ContextEventNotifier`? We don't really care about context level error events for `waitUntilPendingNotificationsDone`, which is used mostly for test.

Reply:
Please note that `eventNotifier` is a `protected` property of `Context`. We should be able to use it as follows in the `TestContext`:

```ts
class TestContext extends Context {
  waitUntilPendingNotificationsDone(timeout?: number) {
    return this.eventNotifier.waitUntilPendingNotificationsDone(timeout);
  }
}
```